src/components/CreateNewRocket.svelte

```diff
@@ -7,33 +7,87 @@
 	import { Label } from '$lib/components/ui/label/index.js';
 	import * as Alert from '@/components/ui/alert';
 	import Checkbox from '@/components/ui/checkbox/checkbox.svelte';
-	import { getRocketURL } from '@/helpers';
+	import { Rocket, ZapRocketNamePurchase } from '@/event_helpers/rockets';
+	import { formatSats, getRocketURL } from '@/helpers';
 	import { ndk } from '@/ndk';
 	import { BitcoinTipTag } from '@/stores/bitcoin';
 	import { currentUser, devmode, mainnet } from '@/stores/session';
 	import { NDKEvent } from '@nostr-dev-kit/ndk';
 	import type NDKSvelte from '@nostr-dev-kit/ndk-svelte';
-	import type { NDKEventStore } from '@nostr-dev-kit/ndk-svelte';
 	import { Info, Terminal } from 'lucide-svelte';
 	import { onDestroy } from 'svelte';
-	import { writable } from 'svelte/store';
+	import { derived, writable } from 'svelte/store';
+	import PayRocketName from './PayRocketName.svelte';
+	import {
+		BUY_ROCKET_NAME_ZAPPED_PUBKEY,
+		BUY_ROCKET_NAME_ZAPPED_EVENT,
+		BUY_ROCKET_NAME_FEE
+	} from '@/consts';
 
-	let rockets: NDKEventStore<NDKEvent> | undefined;
-	const rocketsStore = writable<NDKEvent[]>([]);
-	let name: string = '';
-	$: nameInvalid = true;
-	$: nameError = '';
+	let rocketEvents = $ndk.storeSubscribe([{ kinds: [31108 as number] }], { subId: 'rockets' });
+	let zaps = $ndk.storeSubscribe(
+		[
+			{
+				'#a': [`31108:${BUY_ROCKET_NAME_ZAPPED_PUBKEY}:${BUY_ROCKET_NAME_ZAPPED_EVENT}`],
+				kinds: [9735]
+			}
+		],
+		{
+			subId: BUY_ROCKET_NAME_ZAPPED_EVENT + '_zaps'
+		}
+	);
 
-	rockets = $ndk.storeSubscribe([{ kinds: [31108 as number] }], { subId: 'rockets' });
 	onDestroy(() => {
-		rockets?.unsubscribe();
+		rocketEvents?.unsubscribe();
+		zaps?.unsubscribe();
 	});
-	$: if (rockets) {
-		rockets.subscribe((events) => {
-			rocketsStore.set(events);
+
+	const rockets = writable<Rocket[]>([]);
+
+	$: if (rocketEvents) {
+		rocketEvents.subscribe((events) => {
+			rockets.set(events.map((e) => new Rocket(e)));
```

Comment:
This should probably use derived store instead of doing it this way?

Reply:
Yes, I will modify it later.